goss-testing/automated/print_goss_json_results

```diff
@@ -0,0 +1,218 @@
+If node name is specified, it will be included in the test result stanzas.
+
+Exit codes:
+                            default         --override-rc
+All tests passed            0               0
+At least one test failed    3               0
+No passed or failed tests   4               0
+Usage error                 2               2
+Other error                 1               1
+
+Script log file is in directory /opt/cray/tests/install/logs
+"""
+
+import argparse
+import colorama
+import json
+import logging
+import os
+import requests
+import sys
+import traceback
+
+logger = logging.getLogger(__file__)
+logger.setLevel(logging.DEBUG)
+
+myname = sys.argv[0].split('/')[-1]
+# Strip off .py, if present
+if myname[-3:] == ".py":
+    myname = myname[:-3]
+
+# set up logging to file
+logFileDir = "/opt/cray/tests/install/logs"
+# create log directory; it is ok if it already exists
+os.makedirs(logFileDir, exist_ok=True)
+logFilePath = f"{logFileDir}/{myname}.log"
+file_handler = logging.FileHandler(filename=logFilePath)
+file_handler.setLevel(os.environ.get("PRINT_GOSS_FILE_LOG_LEVEL", logging.DEBUG))
+logger.addHandler(file_handler)
+
+err_text = colorama.Fore.LIGHTRED_EX
+warn_text = colorama.Fore.LIGHTYELLOW_EX
+ok_text = colorama.Fore.LIGHTGREEN_EX
+reset_text = colorama.Style.RESET_ALL
+
+def stderr_print(s):
+    sys.stderr.write(f"{s}\n")
+    sys.stderr.flush()
+
+def warn(msg):
+    stderr_print(f"{warn_text}WARNING: {msg}{reset_text}")
+
+def err(msg):
+    stderr_print(f"{err_text}ERROR: {msg}{reset_text}")
+
+def debug_log_values(**kwargs):
+    logger.debug(" ".join([ f"{k}={v}" for (k, v) in kwargs.items() ]))
+
+class ScriptException(Exception):
+    pass
+
+def parse_args():
+    parser = argparse.ArgumentParser(description="Print JSON-format Goss test results with pretty colors")
+    parser.add_argument("--input", default="-", metavar="input_source", help="Source file or URL for test results. Defaults to stdin.")
+    parser.add_argument("--node_name", help="Node name to include in result stanzas ")
+    parser.add_argument("--override-rc", action="store_true", dest="override_rc", help="Make script exit code ignore test results")
+    args = parser.parse_args()
+    input_source, node_name, override_rc = args.input, args.node_name, args.override_rc
+    debug_log_values(input_source=input_source, node_name=node_name, override_rc=override_rc)
+    return input_source, node_name, override_rc
+
+def read_and_decode_json(input_file):
+    if input_file == "-":
+        input = sys.stdin.read()
+    else:
+        try:
+            with open(input_file, "rt") as infile:
+                input = infile.read()
+        except OSError:
+            logger.error(traceback.format_exc())
+            traceback.print_exc()
+            raise ScriptException(f"Problem reading input file {input_file}")
+    try:
+        return json.loads(input)
+    except json.decoder.JSONDecodeError:
```

Comment:
Might want a general except here too in case you get something like an EPERM or non OSErrror/jsondecode error(s) if only to clean up the stack trace dumped to users of the script.

Reply:
I made changes to address this. Please mark this resolved if you agree (I'd do it myself, but I like to make sure the person who commented is in agreement).